aptos-move/e2e-move-tests/src/tests/offer_rotation_capability.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, MoveHarness};
+use aptos_cached_packages::aptos_stdlib;
+use aptos_crypto::SigningKey;
+use aptos_language_e2e_tests::account::Account;
+use aptos_types::{
+    account_address::AccountAddress,
+    account_config::{AccountResource, CORE_CODE_ADDRESS},
+};
+use move_core_types::parser::parse_struct_tag;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+struct RotationCapabilityOfferProofChallengeV2 {
+    account_address: AccountAddress,
+    module_name: String,
+    struct_name: String,
+    chain_id: u8,
+    sequence_number: u64,
+    source_address: AccountAddress,
+    recipient_address: AccountAddress,
+}
+
+#[test]
+fn offer_rotation_capability_test() {
+    let mut harness = MoveHarness::new();
+    let offerer_account = harness.new_account_with_key_pair();
+    let delegate_account = harness.new_account_with_key_pair();
+
+    offer_rotation_capability_v2(&mut harness, &offerer_account, &delegate_account);
+    revoke_rotation_capability(&mut harness, &offerer_account, *delegate_account.address());
+}
+
+pub fn offer_rotation_capability_v2(
+    harness: &mut MoveHarness,
+    offerer_account: &Account,
+    delegate_account: &Account,
+) {
+    let rotation_capability_proof = RotationCapabilityOfferProofChallengeV2 {
+        account_address: CORE_CODE_ADDRESS,
+        module_name: String::from("account"),
+        struct_name: String::from("RotationCapabilityOfferProofChallengeV2"),
+        chain_id: 4,
```

Comment:
There is a constant for mainnet defined in `types/src/chain_id.rs`: `ChainId::mainnet().id()`